lib/internal/http2/core.js

```diff
@@ -707,7 +707,9 @@ class Http2Session extends EventEmitter {
     const setupFn = setupHandle(this, socket, type, options);
     if (socket.connecting) {
       this[kState].connecting = true;
-      socket.once('connect', setupFn);
+      const connectEvent =
+        socket instanceof tls.TLSSocket ? 'secureConnect' : 'connect';
+      socket.once(connectEvent, setupFn);
```

Comment:
Is it possible to add a test for this?

Reply:
There are already tests for it that have been flaky, this should fix those flaky tests. One of the existing ones could be modified to check that a `secureConnect` handler has been registered.

Reply:
@jasnell Can you point me to one? I volunteer to try and make it fail/pass consistently :)

Reply:
http2-create-client-secure-session is one, which should be fixed by this PR.
I've updated it to include the check for `secureConnect`